rdtools/degradation.py

```diff
@@ -267,31 +291,70 @@ def degradation_year_on_year(energy_normalized, recenter=True,
     df['usage_of_points'] = df.yoy.notnull().astype(int).add(
                 df_right.yoy.notnull().astype(int), fill_value=0)
 
-    calc_info = {
-        'YoY_values': yoy_result,
-        'renormalizing_factor': renorm,
-        'usage_of_points': df['usage_of_points']
-    }
-
     if not len(yoy_result):
         raise ValueError('no year-over-year aggregated data pairs found')
 
     Rd_pct = yoy_result.median()
 
-    # bootstrap to determine 68% CI and exceedance probability
-    n1 = len(yoy_result)
-    reps = 10000
-    xb1 = np.random.choice(yoy_result, (n1, reps), replace=True)
-    mb1 = np.median(xb1, axis=0)
-
-    half_ci = confidence_level / 2.0
-    Rd_CI = np.percentile(mb1, [50.0 - half_ci, 50.0 + half_ci])
-
-    P_level = np.percentile(mb1, 100.0 - exceedance_prob)
-
-    calc_info['exceedance_level'] = P_level
-
-    return (Rd_pct, Rd_CI, calc_info)
+    if uncertainty_method == 'simple_bootstrap':  # If we need the full results
+        calc_info = {
+            'YoY_values': yoy_result,
+            'renormalizing_factor': renorm,
+            'usage_of_points': df['usage_of_points']
+        }
+
+        # bootstrap to determine 68% CI and exceedance probability
+        n1 = len(yoy_result)
+        reps = 10000
+        xb1 = np.random.choice(yoy_result, (n1, reps), replace=True)
+        mb1 = np.median(xb1, axis=0)
+
+        half_ci = confidence_level / 2.0
+        Rd_CI = np.percentile(mb1, [50.0 - half_ci, 50.0 + half_ci])
+
+        P_level = np.percentile(mb1, 100.0 - exceedance_prob)
+
+        calc_info['exceedance_level'] = P_level
+
+        return (Rd_pct, Rd_CI, calc_info)
+
+    elif uncertainty_method == 'circular_block_bootstrap':
+        # Number of bootstrap repetitions
+        reps = 1000
+
+        # Construct degradation trend time series
+        N = len(energy_normalized)
+        numeric_index = np.arange(N)
+        days_per_index = \
+            (energy_normalized.dt.iloc[-1] - energy_normalized.dt.iloc[0]).days / N
+        degradation_trend = 1 + (Rd_pct / 100 / 365.24 * numeric_index
```

Comment:
We should figure this one out before merging

Reply:
Well spotted! Yes it does, but that is easily sorted out by doing the same change wherever 365.24 is used in soiling.py.

Reply:
Though, I wonder, would it make sense to use the actual [tropical year](https://www.timeanddate.com/astronomy/tropical-year.html), i.e. ~8766 hours, as the reference year length when calculating the degradation rate? In my understanding, this would give a more strictly correct degradation rate. (The current estimate is only off by 0.066 %. But accumulating over a 20 year period, that would amount to a performance offset of 1.3 %, which may not be negligible.) Perhaps consider in a future PR?